addons/l10n_fr_pdp_reports/models/pdp_flow.py

```diff
@@ -0,0 +1,341 @@
+    )
+    payload = fields.Binary(
+        string="Payload",
+        attachment=True,
+        help="XML payload that is ultimately sent to the PDP API.",
+    )
+    payload_filename = fields.Char(string="Payload Filename")
+    transport_identifier = fields.Char(
+        string="Transport Identifier",
+        help="Identifier returned by the PDP transport API.",
+    )
+    transport_status = fields.Char(
+        string="Transport Status",
+        help="Raw status returned by the PDP transport API.",
+    )
+    transport_message = fields.Text(
+        string="Transport Message",
+        help="Additional message or error returned by the PDP transport API.",
+    )
+    currency_id = fields.Many2one(
+        "res.currency",
+        string="Currency",
+        required=True,
+        help="Currency of the aggregated transactions included in the payload.",
+    )
+    document_type = fields.Selection(
+        selection=[("sale", "Sale"), ("refund", "Refund")],
+        string="Document Type",
+        default="sale",
+        required=True,
+        help="Indicates whether the flow covers sales or refund transactions.",
+    )
+    transaction_type = fields.Selection(
+        selection=[
+            ("b2c", "B2C Domestic"),
+            ("international", "International B2B"),
+        ],
+        string="Transaction Type",
+        default="b2c",
+        required=True,
+        help="Identify whether the flow reports domestic B2C or international transactions.",
+    )
+    batch_index = fields.Integer(
+        string="Batch Index",
+        default=1,
+        help="Allows splitting large reporting periods into multiple payloads.",
+    )
+    last_send_datetime = fields.Datetime(
+        string="Last Send On",
+        help="Datetime of the most recent send attempt to the PDP transport.",
+    )
+    attempt_count = fields.Integer(
+        string="Send Attempts",
+        default=0,
+        help="Number of times this flow was sent to the PDP transport.",
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        default=lambda self: self.env.company,
+    )
+    move_ids = fields.Many2many(
+        "account.move",
+        "l10n_fr_pdp_flow_move_rel",
+        "flow_id",
+        "move_id",
+        string="Source Moves",
+        help="Invoices that produced this flow.",
+    )
+
+    def _get_default_name(self):
+        """Generate a readable default name."""
+        return _("Flow %s") % fields.Date.context_today(self)
+
+    @api.model
+    def _selection_flow_type(self):
+        return [
+            ("flux10_en16931", _("Flux 10 - EN16931 Invoice")),
+            ("flux10_en16931_extended", _("Flux 10 - EN16931 Extended Invoice")),
+            ("flux10_cii", _("Flux 10 - CII France CIUS Invoice")),
+            ("flux10_cii_extended", _("Flux 10 - CII France CIUS Extended Invoice")),
+        ]
+
+    def action_build_payload(self):
```

Comment:
This is not really used as an action right ? let's do a private function _build_payload instead what do you think ? and or put everything in _build_payload_values ?